.travis/before_install_xenial.sh

```diff
@@ -48,16 +48,13 @@ export PATH=$PATH:$CUDA_HOME/bin
 
 # Set Compiler Config
 sudo update-alternatives --remove-all gcc
-sudo update-alternatives --remove-all g++
-
 sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 10
 sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-6 20
-
-sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-5 10
-sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-6 20
-
 sudo update-alternatives --install /usr/bin/cc cc /usr/bin/gcc 30
 sudo update-alternatives --set cc /usr/bin/gcc
 
+sudo update-alternatives --remove-all g++
+sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-5 10
+sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-6 20
 sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/g++ 30
 sudo update-alternatives --set c++ /usr/bin/g++
```

Comment:
I was expecting to see an extra piece of logic here to avoid installing CUDA at all when `CUDA_ENABLED=OFF`; otherwise you might be under the impression that CUDA-less builds are working while in reality they might not.

Reply:
updated